src/target.d

```diff
@@ -153,7 +153,12 @@ struct Target
      */
     extern (C++) static uint fieldalign(Type type)
     {
-        return type.alignsize();
+        const size = type.alignsize();
+
+        if ((global.params.is64bit || global.params.isOSX) && (size == 16 || size == 32))
+            return size;
+
+        return (8 < size) ? 8 : size;
```

Comment:
What's the reason for this upper bound of 8/32 bytes (why not allow 64 for cache-line aligned fields etc.)? And I find the name `size` for actual alignment rather confusing.